### PR TITLE
[Bug fix] Configure container project

### PR DESCRIPTION
**Pull request by dilin-MS**
1. [Create c_cpp_properties.json file when configure container environment](https://dev.azure.com/mseng/VSIoT/_workitems/edit/1621581)
2. [Fix typo in build script](https://dev.azure.com/mseng/VSIoT/_workitems/edit/1621584)
3. Code Refactor: Reuse code to generate c_cpp_properties.json file for different OS

### CODE REVIEW

File: src/Models/ArduinoDeviceBase.ts [hunk 1]
```diff
@@ -160,20 +160,46 @@ export abstract class ArduinoDeviceBase implements Device {
   abstract get version(): string;
 
   private async writeCppPropertiesFile(
-      rootPath: string, boardId: string, type: ScaffoldType,
-      cppPropertiesTemplateFileName: string,
-      cppPropertiesFilePath: string): Promise<void> {
+      boardId: string, type: ScaffoldType, plat: string): Promise<void> {
```
Comment (eriolchan) [inline src/Models/ArduinoDeviceBase.ts]: It is better to use full name as platform for easy to understand

File: src/Models/ArduinoDeviceBase.ts [hunk 2]
```diff
@@ -160,20 +160,46 @@ export abstract class ArduinoDeviceBase implements Device {
...
+    }
+
     const cppPropertiesTemplateFilePath =
         this.extensionContext.asAbsolutePath(path.join(
             FileNames.resourcesFolderName, FileNames.templatesFolderName,
             boardId, cppPropertiesTemplateFileName));
     const propertiesContent =
-        fs.readFileSync(cppPropertiesTemplateFilePath).toString();
+        await FileUtility.readFile(type, cppPropertiesTemplateFilePath);
+    const propertiesContentString = propertiesContent.toString();
```
Comment (eriolchan) [inline src/Models/ArduinoDeviceBase.ts]: I have seen many code logic as this to replace version value in file in the past few reviews, can we make it a common function if we frequently use it?

Comment (dilin-MS) [reply src/Models/ArduinoDeviceBase.ts]: Yes. This is **the** common function.

Comment (dilin-MS) [reply src/Models/ArduinoDeviceBase.ts]: Yes. This is the common function.

Comment (dilin-MS) [reply src/Models/ArduinoDeviceBase.ts]: > I have seen many code logic as this to replace version value in file in the past few reviews, can we make it a common function if we frequently use it?

Yes. This is the common function.

File: src/Models/ArduinoDeviceBase.ts (hunk 1 above)
Comment (dilin-MS) [reply src/Models/ArduinoDeviceBase.ts]: Will fix.

**Pull request merged**